packages/solana-contrib/src/transaction/TransactionReceipt.ts

```
@@ -96,7 +101,18 @@ export class TransactionReceipt {
    * @param network
    * @returns
    */
-  generateSolanaExplorerLink(cluster: Cluster = "mainnet-beta"): string {
-    return `https://explorer.solana.com/tx/${this.signature}?cluster=${cluster}`;
+
+  generateTXLink(
+    cluster: Cluster = "mainnet-beta",
+    explorerType: string = ExplorerType.SOLANA_EXPLORER
+  ): string {
+    switch (explorerType) {
+      case ExplorerType.SOLANA_EXPLORER:
+        return `https://explorer.solana.com/tx/${this.signature}?cluster=${cluster}`;
+      case ExplorerType.SOLSCAN:
+        return `https://solscan.io/tx/${this.signature}?cluster=${cluster}`;
+      default:
+        return "";
```

Please throw an Error instead.

---

Thanks for your feedback, done it!